diff --git a/test/base/CWToken.complex.test.ts b/test/base/CWToken.complex.test.ts
index 680939d..8737261 100644
--- a/test/base/CWToken.complex.test.ts
+++ b/test/base/CWToken.complex.test.ts
@@ -273,7 +273,7 @@ describe("Contract 'CWToken' - Premintable, Freezable & Restrictable scenarios",
 
   describe("Frozen and premint balances", async () => {
     let timestamp: number;
-    before(async () => {
+    beforeEach(async () => {
       timestamp = (await getLatestBlockTimestamp()) + 100;
     });
     it("Transfer to purpose account - test 5 with release awaiting", async () => {
@@ -505,7 +505,7 @@ describe("Contract 'CWToken' - Premintable, Freezable & Restrictable scenarios",
 
   describe("Premint and restricted balances", async () => {
     let timestamp: number;
-    before(async () => {
+    beforeEach(async () => {
       timestamp = (await getLatestBlockTimestamp()) + 100;
     });
     it("Transfer to purpose account - test 5 with release awaiting", async () => {
@@ -769,7 +769,7 @@ describe("Contract 'CWToken' - Premintable, Freezable & Restrictable scenarios",
 
   describe("Frozen, restricted and premint balances", async () => {
     let timestamp: number;
-    before(async () => {
+    beforeEach(async () => {
       timestamp = (await getLatestBlockTimestamp()) + 100;
     });
     it("Transfer to purpose account - test 5 with release awaiting", async () => {
@@ -1299,7 +1299,7 @@ describe("Contract 'CWToken' - Premintable, Freezable & Restrictable scenarios",
 
   describe("Premint balance only, no frozen balance or restricted balance", async () => {
     let timestamp: number;
-    before(async () => {
+    beforeEach(async () => {
       timestamp = await getLatestBlockTimestamp() + 100;
     });
     it("Transfer to purpose account with release awaiting - test 5", async () => {
